### PR TITLE
fix(schema): don't create empty resolver objects

### LINKED ISSUE #2070: "Error: 'Subscription' defined in resolvers, but not in schema" when subscriptions are disabled
* **Module**: schema-plugin
* **Version**: 0.16.2
* **Node.js / npm versions:**
* Example/snippet of the model that is causing issue

Subscription disabled with
```
crud: {
      create: false,
      subCreate: false,
      subUpdate: false,
      subDelete: false
    },
```

### PR DESCRIPTION
**Pull request by craicoverflow**
Fixes #2070 

## Verification

1. Disable all subscriptions, either globally through `buildGraphbackAPI` or per model in the `@model` annotation.
Start the server - it should work. Debug the `resolvers` object created by Graphback, there should be no Subscription resolver object.

### CODE REVIEW

**Review by wtrocki — APPROVED** [review]

File: packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts [hunk 1]
```diff
@@ -421,53 +417,70 @@ export class SchemaCRUDPlugin extends GraphbackPlugin {
...
 
-    if (model.crudOptions.create && model.crudOptions.subCreate) {
-      this.addCreateSubscriptionResolver(modelType, subscriptionObj)
-    }
-    if (model.crudOptions.update && model.crudOptions.subUpdate) {
-      this.addUpdateSubscriptionResolver(modelType, subscriptionObj)
-    }
-    if (model.crudOptions.delete && model.crudOptions.subDelete) {
-      this.addDeleteSubscriptionResolver(modelType, subscriptionObj)
+    const createSubEnabled = model.crudOptions.create && model.crudOptions.subCreate;
```
Comment (wtrocki) [inline packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts:466]: I would suggest to remove create from it. This is use case we going to have - we want to have just subscriptions enabled for the object but no create.

Good timing

Comment (craicoverflow) [reply packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts:466]: Sure!

File: packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts [hunk 2]
```diff
@@ -421,53 +417,70 @@ export class SchemaCRUDPlugin extends GraphbackPlugin {
...
    */
-  protected addQueryResolvers(model: ModelDefinition, queryObj: IObjectTypeResolver) {
-    if (model.crudOptions.findOne) {
-      this.addFindOneQueryResolver(model, queryObj)
-    }
-    if (model.crudOptions.find) {
-      this.addFindQueryResolver(model, queryObj)
+  protected addQueryResolvers(model: ModelDefinition, resolvers: IResolvers) {
+    if (model.crudOptions.findOne || model.crudOptions.find) {
+      resolvers.Query = {} as IObjectTypeResolver
```
Comment (machi1990) [inline packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts:424]:
```suggestion
      resolvers.Query =  (resolvers.Query || {}) as IObjectTypeResolver
```
?

File: packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts [hunk 3]
```diff
@@ -421,53 +417,70 @@ export class SchemaCRUDPlugin extends GraphbackPlugin {
...
-      this.addCreateMutationResolver(model, mutationObj)
-    }
-    if (model.crudOptions.update) {
-      this.addUpdateMutationResolver(model, mutationObj)
-    }
-    if (model.crudOptions.delete) {
-      this.addDeleteMutationResolver(model, mutationObj)
+  protected addMutationResolvers(model: ModelDefinition, resolvers: IResolvers) {
+    if (model.crudOptions.create || model.crudOptions.update || model.crudOptions.delete) {
+      resolvers.Mutation = {} as IObjectTypeResolver
```
Comment (machi1990) [inline packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts:443]:
```suggestion
      resolvers.Mutation = (resolvers.Mutation || {}) as IObjectTypeResolver
```
?

File: packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts [hunk 4]
```diff
@@ -421,53 +417,70 @@ export class SchemaCRUDPlugin extends GraphbackPlugin {
...
-      this.addUpdateSubscriptionResolver(modelType, subscriptionObj)
-    }
-    if (model.crudOptions.delete && model.crudOptions.subDelete) {
-      this.addDeleteSubscriptionResolver(modelType, subscriptionObj)
+    const createSubEnabled = model.crudOptions.create && model.crudOptions.subCreate;
+    const updateSubEnabled = model.crudOptions.update && model.crudOptions.subUpdate;
+    const deleteSubEnabled = model.crudOptions.delete && model.crudOptions.subDelete;
+
+    if (createSubEnabled || updateSubEnabled || deleteSubEnabled) {
+      resolvers.Subscription = {} as IObjectTypeResolver;
```
Comment (machi1990) [inline packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts:471]:
```suggestion
      resolvers.Subscription = (resolvers.Subscription || {}) as IObjectTypeResolver;
```
?

**Review by machi1990 — CHANGES_REQUESTED** [review]
There are CI failures which seem to be related to the comments / suggestions inline. 

https://github.com/aerogear/graphback/pull/2073/checks?check_run_id=1133210899

**Review by machi1990 — APPROVED** [review]
Comments addressed! Thank you for the quick fix.

@evanshortiss would love this :-)

**Pull request merged**